marker/v2/builders/layout.py

```diff
@@ -90,16 +103,38 @@ def merge_blocks(self, document_pages: List[PageGroup], provider_page_lines: Pdf
                     document_page.add_full_block(line)
                     nearest_block = document_page.children[min_dist_idx]
                     nearest_block.add_structure(line)
+                    nearest_block.polygon = nearest_block.polygon.merge([line.polygon])
+                    nearest_block.text_extraction_method = "pdftext"
                     assigned_line_idxs.add(line_idx)
-                    for span in spans:
+                    for span in line_spans[line_idx]:
                         document_page.add_full_block(span)
                         line.add_structure(span)
 
             for line_idx in provider_line_idxs.difference(assigned_line_idxs):
-                line, spans = provider_lines[line_idx]
+                line = provider_lines[line_idx]
                 document_page.add_full_block(line)
                 text_block = document_page.add_block(Text, polygon=line.polygon)
+                document_page.add_structure(text_block)
+                text_block.text_extraction_method = "pdftext"
                 text_block.add_structure(line)
-                for span in spans:
+                for span in line_spans[line_idx]:
                     document_page.add_full_block(span)
                     text_block.add_structure(span)
+
+    def check_layout_coverage(
+        self,
+        document_page: PageGroup,
+        provider_lines: List[Line],
+        coverage_threshold=0.6
```

Comment:
This might be high, but I haven't tested it.  Setting it a bit too low is probably fine

Reply:
reduced!